### ISSUE TITLE
Privilege for Minimap, Recipe to get Minimap/Radar

### ISSUE THREAD

**Issue by MarkuBu** [minetest/minetest#5016]
The minimap is a nice feature, but it also can change the gameplay fundamentally.

So it would be nice to have (not sure if this is possible now):
1. Privs for minimap and radar to disable both separately 
2. Lua control to turn on/off the minimap and the radar and set the zoom level.

This would make it possible to create recipes to craft an item to show the minimap or the radar and only shows them if the item is wielded

**Comment by rubenwardy** [comment]
This should be at minetest_game

**Comment by rubenwardy** [comment]
You can use lua to disable the minimap and set modes you can use

**Comment by MarkuBu** [comment]
hud_set_flags(flags) only enables/disables the minimap. I can't select the zoom level or if I want to see the minimap or the radar. At least I can't find any function for it.

I also can't find privs for the minimap and radar

What if I want to create a server with a minimap, but without radar, because radar makes the game to easy?

So why is this closed? I'm pretty sure this can't be done in MTG

**Comment by rubenwardy** [comment]
Privs should be a subgame feature imo. There is already an issue about better minimap apis

**Comment by MarkuBu** [comment]
Well, but is it possible to set different privs for minimap and radar from a mod? Is it possible to turn on radar instesd of the minimap from a mod?

**Comment by rubenwardy** [comment]
It's currently only possible to enable or disable the minimap as a whole. This does not mean we should add privileges however - it means we should add it to the hud flags. It's possible to make a privilege that gives you both minimap and radar, however until #4713 is merged you'll need to poll privileges.

Doing it a hud flag makes it easier for subgames to define map items

See #4378 #3051 #3747

tl;dr: the proper fix is to add the hud flags / ability to choose what states are available, then allow mods to choose whether it's privilege or item based

**Comment by nerzhul** [comment]
@rubenwardy please note if there is a priv here it will be like noclip, quasi impossible to anti cheat

**Comment by rubenwardy** [comment]
Nope, it changes nothing. You can override hud settings in the same way as privs like noclip/fly. Also, it's worse the noclip as the server will have no idea if you're cheating

**Comment by nerzhul** [comment]
@rubenwardy exactly what is said

**Comment by est31** [comment]
This is something for clientside lua.

**Comment by est31** [comment]
Its a dupe I think of #4378 (at least the part that doesn't belong into mtgame).

**Comment by paramat** [comment]
Related MTG issue https://github.com/minetest/minetest_game/issues/1787
